components/Root.client.js

```diff
@@ -9,13 +9,14 @@ import { supabase } from '../libs/initSupabase'
 export default function Root() {
   useEffect(() => {
     const { data: authListener } = supabase.auth.onAuthStateChange(
-      (event, session) => {
-        fetch('/api/auth', {
+      async (event, session) => {
+        await fetch('/api/auth', {
           method: 'POST',
           headers: new Headers({ 'Content-Type': 'application/json' }),
           credentials: 'same-origin',
           body: JSON.stringify({ event, session }),
         }).then(res => res.json())
+        if (event === 'SIGNED_IN') window.location.reload()
```

Comment:
@leerob any ideas how to do this better? From what I understand we need to reload after setting the cookie because the components are rendered on the server. Or do you know of another way?